config/interpolate_funcs.go

```diff
@@ -1657,3 +1663,74 @@ func interpolationFuncAbs() ast.Function {
 		},
 	}
 }
+
+// interpolationFuncRsaDecrypt implements the "rsadecrypt" function that does
+// RSA decryption.
+func interpolationFuncRsaDecrypt() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeString, ast.TypeString},
+		ReturnType: ast.TypeString,
+		Callback: func(args []interface{}) (interface{}, error) {
+			s := args[0].(string)
+			key := args[1].(string)
+
+			b, err := base64.StdEncoding.DecodeString(s)
+			if err != nil {
+				b = []byte(s)
+			}
+
+			block, _ := pem.Decode([]byte(key))
+			if block == nil {
+				return "", fmt.Errorf("Failed to read key %q: no key found", key)
+			}
+			if block.Headers["Proc-Type"] == "4,ENCRYPTED" {
```

Comment:
It seems like you take a different approach to load the key in this function vs. the one below.

In this case we check to see if the key is encrypted and, I think, implicitly require it to be RSA by using `ParsePKCS1PrivateKey`. In the other function we don't check for an encrypted key and we use `ParsePXIXPublicKey` which I think accepts DSA and ECDSA keys too, which would then crash on the `.(*rsa.PublicKey)` type assertion underneath.

Does this work differently than I think here, or is there a reason why the two should be different?